src/object/tcb.c

```diff
@@ -1328,6 +1328,11 @@ exception_t decodeSetSchedParams(cap_t cap, word_t length, extra_caps_t excaps,
             current_syscall_error.type = seL4_IllegalOperation;
             return EXCEPTION_SYSCALL_ERROR;
         }
+        if (isBlocked(tcb) && !sc_released(sc)) {
```

Comment:
Yup, they found that all of the IPC code assumed that when you unblock a receiver or sender on IPC there is either no SC or the SC is configured, ready, and released. This is known to be true at the time that any TCB initially blocks on an IPC object and is even true when the SC is removed. They were also able to show that `refill_update` and `refill_new` would never make an SC that was released become not released, so that also didn't break this. The only way that the assumption could be broken is if a TCB that was blocked on an IPC object had an SC bound to it that wasn't already released.